From: gregkh@suse.de
Subject: kobject warning stuff

Good for development, don't push it to mainline as there are too many false positives.


Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 lib/kobject.c |    7 +++++++
 1 file changed, 7 insertions(+)

--- gregkh-2.6.orig/lib/kobject.c
+++ gregkh-2.6/lib/kobject.c
@@ -127,6 +127,7 @@ EXPORT_SYMBOL_GPL(kobject_get_path);
  */
 void kobject_init(struct kobject * kobj)
 {
+	WARN_ON(atomic_read(&kobj->kref.refcount));
 	kref_init(&kobj->kref);
 	INIT_LIST_HEAD(&kobj->entry);
 	init_waitqueue_head(&kobj->poll);
@@ -363,6 +364,12 @@ void kobject_cleanup(struct kobject * ko
 	kobj->k_name = NULL;
 	if (t && t->release)
 		t->release(kobj);
+	else
+		pr_debug("kobject '%s' does not have a release() function, "
+			"if this is not a directory kobject, it is broken "
+			"and must be fixed.\n",
+			kobj->name);
+
 	if (s)
 		kset_put(s);
 	if (parent) 
